sdk/core/core/inc/az_result.h

```diff
@@ -52,19 +52,16 @@ typedef enum az_result {
   AZ_ERROR_BUFFER_OVERFLOW = AZ_MAKE_ERROR(AZ_CORE_FACILITY, 2),
   AZ_ERROR_OUT_OF_MEMORY = AZ_MAKE_ERROR(AZ_CORE_FACILITY, 3),
   AZ_ERROR_NOT_IMPLEMENTED = AZ_MAKE_ERROR(AZ_CORE_FACILITY, 4),
-  AZ_ERROR_UNEXPECTED_CHAR = AZ_MAKE_ERROR(AZ_CORE_FACILITY, 5),
+  AZ_ERROR_ITEM_NOT_FOUND = AZ_MAKE_ERROR(AZ_CORE_FACILITY, 5),
+  AZ_ERROR_PARSER_UNEXPECTED_CHAR = AZ_MAKE_ERROR(AZ_CORE_FACILITY, 6),
```

Comment:
I would expect this to be AZ_PARSER_FACILITY given our naming patterns of ERRORS

Reply:
For parser we have only one error. Let's do it if we have more parsing errors in the future.